dsa-23au/java-dsa/pswish-pipeline/Homework_problems/week9/PrimsPython.py

```diff
@@ -0,0 +1,92 @@
+class Edge:
+    def __init__(self, source, destination, weight):
+        # Set up the Edge Node
+        self.source = source
+        self.destination = destination
+        self.weight = weight
+
+class Graph:
+    def __init__(self):
+        self.BAG = {}
+
+    def add_edge(self, source, destination, weight):  # add each edge to the bag
+        leading_edge = Edge(source, destination, weight)
+        trailing_edge = Edge(destination, source, weight)
+
+        self.BAG.setdefault(source, []).append(leading_edge)  # set default is a complex form of get()
```

Comment:
what does `setdefault` do for a Python dictionary?

Reply:
It is a built-in function of the dicts class where it is basically a get() method for keys where if the key doesn't exist, this method will create and add the values instead of returning a key error.